### PR TITLE
Fix SourceLookUp for nested access to ignored objects

### PR DESCRIPTION
**Pull request by jeeminso**
## Summary of the changes / Why this improves CrateDB
Fixes the following scenarios:
```
cr> create table t (o object(ignored));                                                                                                           
CREATE OK, 1 row affected  (0.151 sec)
cr> insert into t values ({a=1});                                                                                                                 
INSERT OK, 1 row affected  (0.010 sec)
cr> select o['a']['unknown'] from t;                                                                                                              
+-------------------+
| o['a']['unknown'] |
+-------------------+
|                 1 |  -- should be null
+-------------------+
SELECT 1 row in set (0.003 sec)
```

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/main/java/io/crate/expression/reference/doc/lucene/SourceLookup.java [hunk 1]
```diff
@@ -115,6 +115,9 @@ static Object extractValue(final Map<?, ?> map, List<String> path, int pathStart
                 }
                 return newList;
             } else {
+                if (i + 1 != path.size()) {
+                    return null;
+                }
                 break;
```
Comment (BaurzhanSakhariev) [inline server/src/main/java/io/crate/expression/reference/doc/lucene/SourceLookup.java:121]: I wonder if we can remove `SourceLookup.extractValue()` completely and use `ValueExtractors.fromMap()` instead.

`ValueExtractors.fromMap` internally uses `Maps.getyPath `which seems to cover already this `if (i + 1 == path.size()) {`

PKLookup already uses `ValueExtractors` to deal with parsed source (we provide parsed source, processed by SourceParser in https://github.com/crate/crate/pull/14697)

Comment (BaurzhanSakhariev) [reply server/src/main/java/io/crate/expression/reference/doc/lucene/SourceLookup.java:121]: Forgot to mention, if we use ValueExtractors and it doesn't pass your test, we can fix ValueExtractors itself and such fix not only SourceLookup but PKLookup as well.

Comment (mfussenegger) [reply server/src/main/java/io/crate/expression/reference/doc/lucene/SourceLookup.java:121]: Sounds good to deduplicate some of that, but I'd do that as a follow up. For a fix that gets backported, a small fix like this is ideal

**Review by mfussenegger — APPROVED** [review]

File: server/src/main/java/io/crate/expression/reference/doc/lucene/SourceLookup.java (hunk 1 above)
Comment (BaurzhanSakhariev) [reply server/src/main/java/io/crate/expression/reference/doc/lucene/SourceLookup.java:121]: I'm merging it now then - about to rebase drop-column, SourceParser is actively updated there, let's get this change asap.

**Pull request merged**